### ISSUE TITLE
Simplifying the Examples repository.

### ISSUE THREAD

**Issue by kartikdutt18** [mlpack/examples#66]
As per discussion in #61, The aim of this repo needs to be as simple as possible. The main aim might include :
1. Simplifying the code.
2. Maybe remove CMake, I think this would simplify the repo to the extent that one could user could copy the code, change the dataset path and change preprocessing a bit for his/her dataset and he should be good to go.
Looking forward to your response.

To Do:
- [x] Set Up mlpackbot and labels.
- [x] Update Readme for examples.
- [x] Update Readme for models.
- [x] Close #66 
- [x] Simplify code in examples (partially solved in #55 and #56).
- [x] Simplify the repo by removing cmake completely.
- [x] Remove any examples that are too complex for this repository. ( I think VAE falls into this category). Or maybe we could simplify this too ?

Just a suggestion: For when creating documentation for datasets / models, would be a good idea to add images and tables.

Thanks a lot!!!

**Comment by rcurtin** [comment]
Ah, @kartikdutt18, thanks for opening this. You beat me to it. :-D  (Also maybe worth referencing #61 too perhaps?)

I agree, I think that we should remove the CMake configuration entirely, and provide only very simple compilation instructions.  The intent of the code is to be easily copy-paste-able, so getting the compiler and everything set up is a little bit outside the scope of this repository.

If you like, we can use this as a "master issue" for the transition, keeping track of all the things to be done.  Here are a couple tasks I can think of that still need to be done:

 - Update the README (partially covered by #65)---I'm working on a simple draft now, I should have a PR open shortly.
 - Set up labels and mlpack-bot correctly (handled :)).
 - Simple Makefiles for each example.
 - Remove any examples that are too complex for this repository.

Maybe that's not everything, if anyone can think of any, feel free to add. :smile:

**Comment by kartikdutt18** [comment]
>If you like, we can use this as a "master issue" for the transition, keeping track of all the things to be done. Here are a couple tasks I can think of that still need to be done:
Sure, Thats sounds great. I have added a todo list in the description.

**Comment by zoq** [comment]
So I'm a big fan of notebooks, so what about we rewrite the examples as notebooks; GitHub has a notebook renderer so people can copy/paste the code and on top of that we can provide an easy way to run the examples, no need to worry about installing the correct packages or using the correct compiler flags.

**Comment by shrit** [comment]
@rcurtin are you working on simple makefiles? if not I start working on it.

**Comment by shrit** [comment]
@zoq Are you referring to gitbook? I think the documentation are going to be a readme file in each example.

**Comment by zoq** [comment]
Just the notebook file, @rcurtin had a nice example: https://github.com/ageron/handson-ml.

**Comment by zoq** [comment]
I think what we have works, we don't necessary have to add a bunch of documentation around it.

**Comment by shrit** [comment]
I think documentation will be improved slowly. However,  It would be nice to have a list of examples to add in the near future. Actually, it would be nice if we have a set of examples as described in handson-ml

**Comment by rcurtin** [comment]
I haven't worked on any Makefiles; @shrit if you want to do that please feel free.  Maybe we can keep an `.ipynb` notebook file (`.icppnb`?? :smile:) and have its plaintext equivalent too.  That's what we do at my company (we just have to make sure to keep them in sync).

**Comment by kartikdutt18** [comment]
> So I'm a big fan of notebooks, so what about we rewrite the examples as notebooks; GitHub has a notebook renderer so people can copy/paste the code and on top of that we can provide an easy way to run the examples, no need to worry about installing the correct packages or using the correct compiler flags.
That sounds like a great idea. I can push a PR today for some examples maybe if you get a chance you could take a look.

**Comment by kartikdutt18** [comment]
This can be closed after #72, #74 and #75.

**Comment by kartikdutt18** [comment]
Ahh, we can close this. Thanks everyone!
